File: 2441. Largest Positive Integer That Exists With Its Negative.py
"""
Given an integer array nums that does not contain any zeros, find the largest positive integer k such that -k also exists in the array.

Return the positive integer k. If there is no such integer, return -1.
"""
class Solution(object):
    def findMaxK(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        s = set(nums)
        pairs = [n for n in s if n > 0 and -n in s]
        return -1 if len(pairs) == 0 else max(pairs)

File: 2441. Largest Positive Integer That Exists With Its Negative.c
/*
Given an integer array nums that does not contain any zeros, find the largest positive integer k such that -k also exists in the array.

Return the positive integer k. If there is no such integer, return -1.
*/
int findMaxK(int* nums, int numsSize) {
    int min = -1000;
    int cache[2001]={0};
    int output = -1;
    for(int i = 0; i < numsSize; i++){
        cache[nums[i]-min] = 1;
        if(cache[nums[i]*-1-min] == 1){
            output = abs(nums[i]) > output ? abs(nums[i]) : output;
        }
    }
    return output;
}